src/Mod/TechDraw/App/CosmeticExtension.cpp

```diff
@@ -205,8 +205,7 @@ std::string CosmeticExtension::addCosmeticVertex(const Base::Vector3d& pos, bool
     TechDraw::CosmeticVertex* cv = new TechDraw::CosmeticVertex(tempPos);
     verts.push_back(cv);
     CosmeticVertexes.setValues(verts);
-    std::string result = cv->getTagAsString();
-    return result;
+    return cv->getTagAsString();
```

Comment:
Some may say that having temporary variable allows for better debuggability.

Reply:
True. The commit transforms lvalue into rvalue. For hard optimization flags it makes no difference, for low optimization levels, like `-Og` option, it does.

Reply:
If you want to debug it just change the code and add the intermediate variable with auto. Debug code should not be left in the repository.

Reply:
> Debug code should not be left in the repository.


Why? Is there a rule? _FreeCAD_ contains a lot of debug stuff, #ifdefs, for example.

Reply:
I think this depends a lot on the situation.
In this specific case, I personally don't see why it's important to keep around as return values can also be inspected with debuggers.
By removing these values it simplifies the code to read and parse.

You are correct that FreeCAD contains a lot of debug stuffs, but I'm not sure I follow why that's a reason to keep it here. Maybe those places also needs some clean up work?

Reply:
> You are correct that FreeCAD contains a lot of debug stuffs, but I'm not sure I follow why that's a reason to keep it here. Maybe those places also needs some clean up work?



The reason to write debug code is to debug program easier.


----


I simply do not understand the sentence: "_Debug code should not be left in the repository._" Is it a policy?

Reply:
> The reason to write debug code is to debug program easier.

Sure, but it does not mean that it should be left in the repository as it does not bring any value for the released application. 

> I simply do not understand the sentence: "Debug code should not be left in the repository." Is it a policy?

Is it a policy? Not, at least not yet but there is little to no reason to leave it only for debugging reasons. It is easy enough to add the intermediate line while you are debugging and delete it later. If this is ifdef that is clearly marked as debug code and is not trivial to add - sure it may be worth to keep it.

Reply:
>  It is easy enough to add the intermediate line while you are debugging and delete it later.

To be honest it is a pain in the ass when the reality of development hits you and you have to add 30 such single intermediate lines, and you have to keep them locally forever, and constantly need to stash them or rebase on pulling changes which causes extra rebuilding.